File: sarc/sarc.php
<?php

array_shift($argv);

$text = implode(" ", array_map(fn ($arg) => trim($arg), $argv));

if (!$text) {
  print("USAGE: sarc The text you want to be sarcastic" . PHP_EOL);
  exit(1);
}

$shouldUpperCase = true;
$output = "";

for ($i = 0; $i < strlen($text); $i++) {
  if (ctype_alpha($text[$i])) {
    $shouldUpperCase = !$shouldUpperCase;
  }

  $output .= $shouldUpperCase ? strtoupper($text[$i]) : strtolower($text[$i]);
}

print($output . PHP_EOL);

File: sarc/sarc.go
package main

import (
	"fmt"
	"os"
	"strings"
	"time"
	"unicode"
)

func main() {
	start := time.Now()
	text := strings.Trim(strings.Join(os.Args[1:], ""), " ")

	if len(text) == 0 {
		fmt.Println("USAGE: sarc The text you want to be sarcastic")

		os.Exit(1)
	}

	sarcasticText := ToSarcastic(text)
	fmt.Println(sarcasticText, time.Since(start))
}

func ToSarcastic(text string) string {
	output := []rune(text)
	shouldUpperCase := true

	for idx, character := range text {
		if unicode.IsLetter(character) {
			shouldUpperCase = !shouldUpperCase
		}

		output[idx] = unicode.ToLower(character)
		if shouldUpperCase {
			output[idx] = unicode.ToUpper(character)
		}
	}

	return string(output)
}
